src/modes/edit_message.cc

```diff
@@ -666,7 +668,56 @@ namespace Astroid {
   }
 
   bool EditMessage::set_from (Account * a) {
-    generate_mid_from_account(a);
+    if (draft_msg) {
+      LOG(debug) << "em: set_from called for saved draft";
+       if (a != accounts->get_account_for_address (draft_msg->sender)) {
+         LOG(debug) << "em: set_from called for saved draft with different account";
+
+         ustring fname = draft_msg->fname;
+         path fpath = path(fname.c_str ());
+         if (is_regular_file (fpath)) {
+           boost::filesystem::remove (fpath);
+            ComposeMessage * cmsg = make_draft_message ();
+            ustring new_mid = generate_mid_from_account (a);
+            cmsg->set_id (new_mid);
+//            draft_msg->mid = msg_id;         // propagate new message id to draft_msg
+//            draft_msg->nmmsg->mid = msg_id;
+//            draft_msg->message->message_id = (char*) msg_id.c_str();
+            ThreadView *tv = thread_view;
+            astroid->actions->doit(refptr<Action>(new OnMessageAction (draft_msg->mid, "" /*draft_msg->tid*/,
+              [fname, cmsg, tv] (Db * db, notmuch_message_t * msg) {
+                // first, remove any draft tags
+                for (ustring t : Db::draft_tags) {
+                  notmuch_message_remove_tag (msg, t.c_str ());
+                }
+                // remember remaining (non-draft) tags (if any)
+                notmuch_tags_t *keep_tags = notmuch_message_get_tags(msg);
```

Comment:
Is this struct safe when message is removed?

Reply:
Good point. Should better make a copy.